datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -569,7 +694,9 @@ def warning(self, warning_message, *args, **kwargs):
         # https://github.com/python/cpython/blob/1dbe5373851acb85ba91f0be7b83c69563acd68d/Lib/logging/__init__.py#L368-L369
         if args:
             warning_message = warning_message % args
-        frame = inspect.currentframe().f_back
+        currentframe = inspect.currentframe()
+        assert currentframe is not None
```

Comment:
> how would that happen?

In theory `inspect.currentframe()` *could* return `None`"if getting the current frame is not possible" (quoting the docstring of that function). In practice we're inside a function, so *there will be* a current frame (the one for `warning()`). So I'm using an assertion to help mypy and tell it "hey, I know what I'm doing here — this will *never* be a `None`".

The other option to do this is to use `typing.cast()`, which tells mypy "hey, this is actually *this particular type*". Expect it's less flexible, because you can only do an equivalent of an `assert isinstance(typ, value)` check. Eg you can't `typing.cast('Not None!', value)`. So we'd need to do…

```python
currentframe = typing.cast('<Type of a frame (not sure what that would be?)>', inspect.currentframe())
```

The absolute other alternative is to shut up mypy with a `# type: ignore` here, but honestly the `is not None` assertion is IMO the sanest way to go. (I'm not using it for a runtime side effect, in which case I'd agree we should be raising an appropriate exception.)

More info on casting and assertions: https://mypy.readthedocs.io/en/stable/casts.html

Reply:
(Also I don't think we should be an absolute "no" on keeping assertions in production code when it refers to theoretical states that *should not* happen, but that's probably a different topic…)

Reply:
Thanks for the explanation! I'm now a fan of doing `# type: ignore` here. No need to incur the cost of a runtime check that will never fail just to appease a tool, imo